src/main/java/com/forgeessentials/signtools/SignToolsModule.java

```diff
@@ -120,35 +117,45 @@ public void onPlayerInteract(PlayerInteractEvent event)
 
             }
 
-            String[] signText = getFormatted(sign.signText);
+            String[] signText = getUnFormatted(sign.signText);
 
             if (APIRegistry.scripts.runEventScripts(signinteractKey, event.getEntityPlayer(), new SignInfo(event.getEntityPlayer().dimension, event.getPos(), signText, event)))
             {
                 event.setCanceled(true);
             }
 
+            /**
+             * how to use: First line of the sign MUST BE [command] Second line is the command you want to run Third and fourth
+             * lines are arguments to the command.
+             */
             if (allowSignCommands && (event instanceof RightClickBlock))
             {
-                if (signText[0].equals("[command]"))
+                // if (signText[0].trim() == "[command]")
+                if (signText[0].trim().equalsIgnoreCase("[command]"))
                 {
-                    String send = signText[1] + " " + signText[2] + " " + signText[3];
-                    if (send != null && FMLCommonHandler.instance().getMinecraftServerInstance().getCommandManager() != null)
+                    if (PermissionAPI.hasPermission(event.getEntityPlayer(), USESIGNCOMMANDS_PERM))
                     {
-                        FMLCommonHandler.instance().getMinecraftServerInstance().getCommandManager().executeCommand(event.getEntityPlayer(), send);
-                        event.setCanceled(true);
+                        String send = signText[1] + " " + signText[2] + " " + signText[3];
+                        if (send != null && FMLCommonHandler.instance().getMinecraftServerInstance().getCommandManager() != null)
+                        {
+                            FMLCommonHandler.instance().getMinecraftServerInstance().getCommandManager().executeCommand(event.getEntityPlayer(), send);
+                            event.setCanceled(true);
+                        }
                     }
+                    else
```

Comment:
Code Style rules for Forge Essentials requires brackets for every if / else even if it is a one liner.

Reply:
Didn't know, sorry